Code/Framework/AzCore/Tests/Metrics/JsonTraceEventLoggerTests.cpp

```diff
@@ -0,0 +1,485 @@
+        using EventObjectStorage = AZStd::fixed_vector<AZ::Metrics::EventField, 8>;
+        EventArrayStorage eventArray;
+        EventObjectStorage eventObject;
+
+        // Fill out the child array and child object fields
+        for (const auto& [fieldName, fieldValue] : AZStd::views::zip(objectFieldNames, objectFieldValues))
+        {
+            auto AppendArgs = [name = fieldName, &eventArray, &eventObject](auto&& value)
+            {
+                eventArray.push_back(value);
+                eventObject.emplace_back(name, value);
+            };
+
+            AZStd::visit(AppendArgs, fieldValue);
+        }
+
+        // Populate the "args" container to associate with each events "args" field
+        EventObjectStorage argsContainer;
+        argsContainer.emplace_back("string", eventString);
+        argsContainer.emplace_back("int64_t", eventInt64);
+        argsContainer.emplace_back("uint64_t", eventUint64);
+        argsContainer.emplace_back("bool", eventBool);
+        argsContainer.emplace_back("double", eventDouble);
+        argsContainer.emplace_back("array", AZ::Metrics::EventArray(eventArray));
+        argsContainer.emplace_back("object", AZ::Metrics::EventObject(eventObject));
+
+        // Create an byte container stream that allows event logger output to be logged in-memory
+        AZStd::string metricsOutput;
+        auto metricsStream = AZStd::make_unique<AZ::IO::ByteContainerStream<AZStd::string>>(&metricsOutput);
+
+        // Create the trace event logger that logs to the Google Trace Event format
+        AZ::Metrics::JsonTraceEventLogger googleTraceLogger(AZStd::move(metricsStream));
+
+
+        using ResultOutcome = AZ::Metrics::IEventLogger::ResultOutcome;
+
+        // Defer logging until after all threads have been started
+        AZStd::atomic_bool startLogging{};
+
+        auto LogAllEvents = [&startLogging, &googleTraceLogger, &argsContainer](int threadIndex)
+        {
+            // Fake timestamp to use for complete event
+            AZStd::chrono::utc_clock::time_point startThreadTime = AZStd::chrono::utc_clock::now();
+            while (!startLogging)
+            {
+                AZStd::this_thread::yield();
+            }
+
+            // Convert the threadIndex to a string and use that as the "id" value
+            AZStd::fixed_string<32> idString;
+            AZStd::to_string(idString, threadIndex);
+
+            ResultOutcome resultOutcome(AZ::Metrics::IEventLogger::ErrorString("Uninitialized"));
+            {
+                // Record Duration Begin and End Events
+                AZ::Metrics::DurationArgs durationArgs;
+                durationArgs.m_name = "Duration Event";
+                durationArgs.m_cat = "Test";
+                durationArgs.m_args = argsContainer;
+                durationArgs.m_id = idString;
+                resultOutcome = googleTraceLogger.RecordDurationEventBegin(durationArgs);
```

Comment:
do we want to assert a success outcome for these test record events?

Reply:
We EXPECT_TRUE them on the next line.